sys/include/ztimer.h

```diff
@@ -474,8 +474,103 @@ ztimer_now_t _ztimer_now_extend(ztimer_clock_t *clock);
+ *   stays active when timers are set back-to-back (which is the case when the
+ *   first timer's callback sets the second timer), or when they overlap (which
+ *   can be known by starting the second timer and afterwards observing that
+ *   @ref ztimer_is_set or @ref ztimer_remove returns true in a low-priority
+ *   context).
+ *
+ *   In contrast, the clock is not guaranteed to be active if a timer is
+ *   removed and then a second one is started (even if the thread does not
+ *   block between these events), or when an expiring timer wakes up a thread
+ *   that then sets the second timer.
+ *
+ *   If the clock was active, then the difference between the second value and
+ *   the first is then the elapsed time in the clock's unit, **modulo 2³²
+ *   ticks** (or 2⁶⁴ when using the ZTIMER_NOW64 module).
+ *
+ * * A difference between two values (calculated in the usual wrapping way) is
+ *   guaranteed to be exactly the elapsed time (not just modulo 2³²) if there
+ *   exists a single timer that is continuously set while both
+ *   readings are taken (which in particular means that the clock was
+ *   continuously active), **and** the timer is observed to be still set when
+ *   after the second reading an execution context with lower priority than the
+ *   ZTimer interrupt has run. (In particular, this is the case in a thread
+ *   context when interrupts are enabled).
+ *
+ *   For example, this sequence of events will return usable values:
+ *
+ *   * In a thread, a timer is set.
+ *   * Some interrupt fires, and `start = ztimer_now(ZTIMER_MSEC)` is set in
+ *     the handler.
+ *   * The interrupt fires again, and `duration = start -
+ *     ztimer_now(ZTIMER_MSEC)` is stored.
+ *   * Back in the thread context, @ref ztimer_remove on the timer returns
+ *     true.
+ *
+ *     Only now, `duration` can be known to be a duration in milliseconds.
+ *
+ *   (By comparison, if the timer were removed right inside the second
+ *   interrupt, then duration might either be correct, or it might be 5
+ *   milliseconds when really 2³² + 5 milliseconds have elapsed)
+ *
+ *   The requirement of the execution contexts can be **dispensed with, if**
+ *   the set timer is shorter than the wrap-around time of the clock by at
+ *   least the maximum duration the full system is allowed to spend between
+ *   interrupt servicing opportunities. That time varies by setup, but an
+ *   upper bound of 1 minute is conservative enough for system modules to use.
+ *
+ *   For example, this sequence of events will also return usable values:
+ *
+ *   * A mutex is locked, and a timer is set to unlock it on the millisecond
+ *     timer after 1 hour. (This is way less than the wrap-around time of
+ *     around 50 days).
+ *   * The return value of setting the timer is noted as start time.
+ *   * Some interrupt fires, and `ztimer_now()` is taken. Then (still inside
+ *     the ISR), @ref mutex_try_lock is used to test for whether the interrupt
```

Comment:
```suggestion
 *     the ISR), @ref mutex_trylock is used to test for whether the interrupt
```
static test failed here